_dev/src/ts/routing/ScriptHandler.ts

```diff
@@ -28,16 +25,47 @@ export default class ScriptHandler {
     'update-page-post-update': UpdatePagePostUpdate
   };
 
-  private loadScript(routeName: string) {
-    if (this.routesMatching[routeName]) {
-      const pageClass = this.routesMatching[routeName];
-      this.currentScript = new pageClass();
-      this.currentScript.mount();
+  /**
+   * @constructor
+   * @description Initializes the `ScriptHandler` by loading the page script associated with the current route.
+   */
+  constructor() {
+    const currentRoute = routeHandler.getCurrentRoute();
+
+    if (currentRoute) {
+      this.#loadScript(currentRoute);
+    }
+  }
+
+  /**
+   * @private
+   * @param {string} routeName - The name of the route to load his associated script.
+   * @returns void
+   * @description Loads and mounts the page script associated with the specified route name.
+   */
+  #loadScript(routeName: string) {
+    const pageClass = this.#routesMatching[routeName];
+    if (pageClass) {
+      try {
+        this.#currentScript = new pageClass();
+        this.#currentScript.mount();
+      } catch (error) {
+        console.error(`Failed to load script for route ${routeName}:`, error);
+      }
+    } else {
+      console.warn(`No matching page Class found for route: ${routeName}`);
```

Comment:
Is this really an issue if there is no matching script? I would be more in favor of a `debug` call.

Reply:
if you prefer a console.debug instead of .warn I don't see any problem with that.